pkg/remediation/scan.go

```diff
@@ -0,0 +1,259 @@
+package remediation
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	"github.com/Checkmarx/kics/pkg/engine"
+	"github.com/Checkmarx/kics/pkg/kics"
+	"github.com/Checkmarx/kics/pkg/model"
+	"github.com/Checkmarx/kics/pkg/scan"
+	"github.com/open-policy-agent/opa/topdown"
+
+	"github.com/Checkmarx/kics/internal/console/flags"
+	"github.com/Checkmarx/kics/internal/tracker"
+	"github.com/Checkmarx/kics/pkg/engine/source"
+	"github.com/Checkmarx/kics/pkg/parser"
+	buildahParser "github.com/Checkmarx/kics/pkg/parser/buildah"
+	dockerParser "github.com/Checkmarx/kics/pkg/parser/docker"
+	protoParser "github.com/Checkmarx/kics/pkg/parser/grpc"
+	jsonParser "github.com/Checkmarx/kics/pkg/parser/json"
+	terraformParser "github.com/Checkmarx/kics/pkg/parser/terraform"
+	yamlParser "github.com/Checkmarx/kics/pkg/parser/yaml"
+	"github.com/Checkmarx/kics/pkg/utils"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/rs/zerolog/log"
+)
+
+type runQueryInfo struct {
+	payload   model.Documents
+	query     *engine.PreparedQuery
+	inspector *engine.Inspector
+	tmpFile   string
+	files     model.FileMetadatas
+}
+
+// scanTmpFile scans a temporary file against a specific query
+func scanTmpFile(tmpFile, queryID string, remediated []byte) ([]model.Vulnerability, error) {
+	// get payload
+	files, err := getPayload(tmpFile, remediated)
+
+	os.Remove(tmpFile)
```

Comment:
we should check this error and return a warning in case there as an issue, but not break the flow